sql/src/main/java/org/apache/druid/sql/avatica/DruidAvaticaProtobufHandler.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql.avatica;
+
+import com.google.inject.Inject;
+import org.apache.calcite.avatica.remote.LocalService;
+import org.apache.calcite.avatica.remote.Service;
+import org.apache.calcite.avatica.server.AvaticaProtobufHandler;
+import org.apache.druid.guice.annotations.Self;
+import org.apache.druid.server.DruidNode;
+import org.eclipse.jetty.server.Request;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class DruidAvaticaProtobufHandler extends AvaticaProtobufHandler
+{
+  public static final String AVATICA_PATH = "/druid/v2/sql/avatica-pb/";
```

Comment:
I'm wondering if we can use the same path for both Avatica and Json, but accept a query parameter to choose one of them. I think it could be better in terms of maintenance. In `AsyncQueryForwardingServlet`, it seems that you can check the query string. In `DruidAvaticaHandler`, it can have two delegates for JSON and protobuf and calls one of them based on query string in the request.

Reply:
The problem is the drivers. If we want to support JDBC and golang drivers, I'm not sure they would be happy with always appending a query param to the url. The other option I can think of that may work is to auto-detect it when the first letter of the post is `[` or `{` (ie. json), that just seemed a bit dirty to me

Reply:
I see. It sounds reasonable to me.

Reply:
@jihoonson To be clear, do you think I should go the JSON sniffer way instead of having a separate endpoint?

Reply:
No, I think having a separate endpoint makes sense.